Ensure "Add a note to your order" section is styled correctly when disabled

opr opened the pull request:
When the checkout form is submitted, the fields in the address fieldset are disabled. They have an opacity set to give a visual indication of this. The "Add a note to your order" section does not have this same visual indication. (Though the form fields within do get the disabled attribute)

Fixes #3426 

### Screenshots
#### Before
<img src="https://user-images.githubusercontent.com/5656702/99548353-7642bc00-29b0-11eb-903a-c9aa5b128a9f.png" width="400">

#### After
<img width="400" alt="Screenshot 2020-11-18 at 15 04 12" src="https://user-images.githubusercontent.com/5656702/99547391-68406b80-29af-11eb-8aae-7275a7fa715c.png">

### How to test the changes in this Pull Request:

1. Add a product to your cart and go to a page with the checkout block
2. Fill in the required form fields. Leave the "Add a note to your order" unticked.
3. Submit the form
4. Ensure the "Add a note to your order" section looks disabled
5. Restart the checkout process and fill in all required form fields.
5. Tick the box next to "Add a note to your order" and (optionally) fill in the textarea with something
6. Submit the form again and ensure the entire "Add a note to your order" section is disabled, including the textarea.

opr committed:
Pass disabled prop to FormStep in order-notes-step.js

This way we can add the disabled class to the div or fieldset rendered by FormStep.

opr committed:
Add disabled class & style for disabled FormStep components

Allows us to style disabled elements that cannot have the disabled attribute (divs) the same way as disabled fieldsets are.

opr committed:
Update test snapshot for FormStep

This is because we added a class to the div/fieldset element when it is disabled.

Aljullu commented on assets/js/base/components/cart-checkout/form-step/style.scss:
```diff
@@ -14,6 +14,10 @@
 	}
 }
 
+.wc-block-components-checkout-step--disabled {
```
With this new selector, do you think we can remove the one in [line 29](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/fix/3426-disable-add-note-field/assets/js/base/components/cart-checkout/form-step/style.scss#L29)?

opr replied:
You're right - it's no longer needed. Removed.

Aljullu reviewed (approved):
LGTM, good job! :rocket:  I added only one comment about removing a CSS selector that I think is now redundant, but everything else looks good so pre-approving.

opr committed:
Remove redundant selector from form-step styles

Because we now add the --disabled modifier to the class, we no longer need the disabled selector. The new class-name based selector covers both fieldsets and divs.
